agiverse/agent/memory/importance.py

```diff
@@ -0,0 +1,104 @@
+from typing import List, Optional, Union
+from datetime import datetime
+import numpy as np
+import litellm
+import json
+from tenacity import retry, stop_after_attempt, wait_exponential
+from .base import Memory, MemoryStream
+import os
+import logging
+
+logger = logging.getLogger(__name__)
+
+class ImportanceCalculator:
+    def __init__(self, api_key: str = None, model: str = None):
+        self.api_key = api_key
+        self.model = model or os.getenv('IMPORTANCE_MODEL', 'gpt-4o-mini')
+        self.time_weight = 0.3
+        self.relevance_weight = 0.7
+        
+    async def calculate_relevance(self, memory: Memory, current_time: datetime,
+                                related_memories: List[Memory]) -> float:
+        memory_times = [m.created_at for m in related_memories]
+        time_factor = self._calculate_time_decay(memory.created_at, memory_times)
+        relevance_factor = await self._calculate_relevance(memory, related_memories)
+        
+        importance = [
+            min(max(self.time_weight * t + self.relevance_weight * r, 0.0), 1.0)
+            for t, r in zip(time_factor, relevance_factor)
+        ]
+        return importance
+
+    async def calculate_memory_importance(self, memory: Memory) -> float:
+        current_memory = f"Current memory: {memory.content}"
+        prompt = f"""As an expert in evaluating memory importance, analyze the following memory and its context.
```

Comment:
Let's put all prompts in file and use `agent.get_prompt()` to retrieve it so users can change them.